### ISSUE TITLE
cf restage immediately after cf push --no-start does not work

### ISSUE THREAD

**Issue by gdankov** [cloudfoundry/cloud_controller_ng#1584]
## Issue

When we do `cf push --no-start dora` followed by `cf restage dora` we get the following error message:

```
❯ cf restage dora                          
This action will cause app downtime.

Restaging app dora in org o / space s as admin...

Staging app and tracing logs...
App has not finished staging
FAILED
```

We're not sure if this is expected behavior or a bug. If it is expected that you cannot restage an app that hasn't been staged yet, at least the error message can be improved to explain that.

## Steps to Reproduce

```
cf push --no-start dora
cf restage dora
```

## Expected result

The app should be staged

## Current result

It fails with a confusing message

## Possible Fix

Either (1) stage the app if it hasn't been staged before, or (2) print a better message.

/cc @schaefm-ibm

**Comment by aashah** [comment]
Searching slack history, the root cause _seems_ to be around cf-cli 6.27, when a major rework of `cf push` took place. There was some discussion in slack on this very same problem - https://cloudfoundry.slack.com/archives/C032824SM/p1533590564000108 ...but there was no resolution or Tracker story to follow this "regression".

I quoted regression, because it's unclear at this point what the decision was. I reproduced this following your steps, and totally agree that it's confusing (the team often talks about what the meaning of "restage" even means). Let me check with the related teams to see what the plan is for this.
